services/wallet/router/router.go

```diff
@@ -213,12 +224,14 @@ func (r *Router) SuggestedRoutes(ctx context.Context, input *requests.RouteInput
 		r.activeRoutesMutex.Lock()
 		r.activeRoutes = suggestedRoutes
 		r.activeRoutesMutex.Unlock()
-		if suggestedRoutes != nil && err == nil {
+		r.routeCanceledMutex.Lock()
+		if suggestedRoutes != nil && err == nil && !r.routeCanceled {
 			// subscribe for updates
 			for _, path := range suggestedRoutes.Best {
 				err = r.subscribeForUdates(path.FromChain.ChainID)
 			}
 		}
+		r.routeCanceledMutex.Unlock()
```

Comment:
nit: should this be a defer?

Reply:
@alaibe why? There is no possibility of breaking the flow, no return between `r.routeCanceledMutex.Lock()` and `r.routeCanceledMutex.Unlock()`?